src/sockaddr.rs

```diff
@@ -12,24 +12,68 @@ use winapi::shared::ws2ipdef::SOCKADDR_IN6_LH_u;
 /// The address of a socket.
 ///
 /// `SockAddr`s may be constructed directly to and from the standard library
-/// `SocketAddr`, `SocketAddrV4`, and `SocketAddrV6` types.
+/// [`SocketAddr`], [`SocketAddrV4`], and [`SocketAddrV6`] types.
 pub struct SockAddr {
     storage: sockaddr_storage,
     len: socklen_t,
 }
 
+#[allow(clippy::len_without_is_empty)]
 impl SockAddr {
+    /// Create a `SockAddr` from the underlying storage and its length.
+    ///
+    /// The type of the address storage and length passed to the function `init`
+    /// is OS/architecture specific.
+    ///
+    /// # Safety
+    ///
+    /// - The `SockAddr`'s address family must match its contents.
+    /// - The given length of the `SockAddr` must be correct.
+    ///
+    /// # Examples
+    ///
+    #[cfg_attr(target_os = "linux", doc = "```")]
+    #[cfg_attr(not(target_os = "linux"), doc = "```ignore")]
+    /// use std::io;
+    /// use std::mem;
+    ///
+    /// use socket2::{SockAddr, Socket, Domain, Type, Protocol};
+    ///
+    /// # fn main() -> io::Result<()> {
+    /// // Initialise a netlink socket.
+    /// let mut storage: libc::sockaddr_storage = unsafe { mem::zeroed() };
+    /// let address: &mut libc::sockaddr_nl = unsafe { &mut *<*mut _>::cast(&mut storage) };
+    /// address.nl_family = libc::AF_NETLINK as libc::sa_family_t;
+    /// let address = unsafe { SockAddr::new(storage, mem::size_of::<libc::sockaddr_nl>() as _) };
+    ///
+    /// let socket = Socket::new(
+    ///     Domain::from(libc::AF_NETLINK),
+    ///     Type::from(libc::SOCK_DGRAM),
+    ///     Some(Protocol::from(libc::NETLINK_GENERIC)),
+    /// )?;
+    /// socket.bind(&address)?;
+    /// # drop(socket);
+    /// # Ok(())
+    /// # }
+    #[doc = "```"]
```

Comment:
You can just use:

    /// ```

Reply:
I wrote it this way so that syntax highlighting doesn't break in my editor; with the old method, all doc comments below that would be treated as Rust code. Although I just realized that the example itself could have the appropriate `#![cfg(...)]` at the top and use regular triple-backtick on all platforms, so I'll switch to that instead.